lib/response.js

```diff
@@ -1130,6 +1133,43 @@ function sendfile(res, file, options, callback) {
   file.pipe(res);
 }
 
+
+function convertBigIntsToNumbers(obj) {
+  // If the type is a BigInt, convert them to number
+  if (typeof obj === 'bigint') {
+    return Number(obj);
+  }
+
+  // If the input is not an object or array, return it as is
+  if (typeof obj !== 'object' || obj === null) {
+    return obj;
+  }
+
+  // If it's an array, loop over its elements and convert them recursively
+  if (Array.isArray(obj)) {
+    for (var index = 0; index < obj.length; index += 1) {
+      obj[index] = convertBigIntsToNumbers(obj[index]);
+    }
+    return obj;
+  }
+
+  // Otherwise, it's an object, so iterate over its keys
+  for (var key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      var value = obj[key];
+      // If the value is a BigInt, convert it to a number
+      if (typeof value === 'bigint') {
+        obj[key] = Number(value);
```

Comment:
Will this mutate the object that is passed into Express, or is this operating on a copy? May need to be very mindfull bc objects may have setters defined or even the property can have only a getter and no setter at all.

Reply:
We don't want to mutute whatever the user is passing in to `res.send` as that would be very unexpected and can even cause whatever is done with it to change or be unpredictable. For example the app also passed the object to be saved in the db, but then this is changing the values all of a sudden, changing what gets saved in the db -- or a race condition if this if they are both happening async to each other. The object passed to `res.send` may even be frozen.

Reply:
I understood the case, in which when we sent the response first and then saved into DB, it will lose the fully original context,
I should have a copy of this.